File: examples/opencode-kungfu/quickstart/episode.py
# SPDX-License-Identifier: Apache-2.0

import sys

from kungfu.storage import service

runtime_dir = sys.argv[1]
opened = service.episode_begin(
    runtime_dir, title="vendor quickstart", actor="python-host"
)
print(
    service.episode_end(
        runtime_dir,
        episode_id=opened["episode_id"],
        reason="quickstart complete",
    )
)

File: examples/opencode-kungfu/quickstart/episode.mjs
// SPDX-License-Identifier: Apache-2.0

import core from '@kungfu-tech/core';

const runtimeDir = process.argv[2];
if (!runtimeDir) throw new Error('usage: node episode.mjs RUNTIME_DIR');
const kungfu = core.kungfu();
const opened = kungfu.storageEpisodeBeginTyped(runtimeDir, {
  title: 'vendor quickstart',
  actor: 'node-host',
});
console.log(
  kungfu.storageEpisodeCloseTyped(runtimeDir, {
    episode_id: opened.episode_id,
    reason: 'quickstart complete',
  }),
);
